### PR TITLE
new documentation guidelines

### PR DESCRIPTION
**Pull request by kasiakepka**
**Description**

Changes proposed in this pull request:

- Add documentation guidelines for Neighbors team and their repositories
- 
**Related issue(s)**
[
](https://github.com/kyma-project/test-infra/issues/6147)

### CODE REVIEW

File: docs/documentation_guidelines.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Documentation guidelines
+
+1. Each repository should contain an automatically updated index page in `docs` directory.
+2. Repository main `README.md` file should contain a link to the concerned repository index page, where all documentation contents are stored.
```
Comment (NHingerl) [inline docs/documentation_guidelines.md:4]:
```suggestion
2. The repository's main `README.md` file should contain a link to the repository's index page, where all documentation content is stored.
```

File: docs/documentation_guidelines.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+3. Each `.md` file should contain at the beginning a title and a document description in a paragraph below. These two elements of `.md` file will be used to create index page.
```
Comment (NHingerl) [inline docs/documentation_guidelines.md:5]:
```suggestion
3. Each `.md` file should begin with a title and in the paragraph below, a document description . These two elements of `.md` file will be used to create index page.
```

File: docs/documentation_guidelines.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+4. A `docs` directory under the repository root doesn't have to contain documentation related strictly to the one tool only. For example we can have a documentation describing a system comprised of multiple tools which together deliver some service for users.
```
Comment (NHingerl) [inline docs/documentation_guidelines.md:6]: I think we should first say what it should contain, not what it shouldn't contain :D 

```suggestion
4. A `docs` directory under the repository root doesn't have to contain documentation related strictly to the one tool only. For example, we can have a documentation describing a system comprised of multiple tools which together deliver some service for users.
```

File: docs/documentation_guidelines.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+5. Documentation for specific tools and libraries should be maintained in the same directory as a tool or library.
```
Comment (NHingerl) [inline docs/documentation_guidelines.md:7]:
```suggestion
5. Documentation for specific tools and libraries should be maintained in the same directory as the tool or library.
```

File: docs/documentation_guidelines.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+6. If a tool has more complex documentation structure it can have separate `docs` directory under it's root directory.
```
Comment (NHingerl) [inline docs/documentation_guidelines.md:8]:
```suggestion
6. If a tool has a more complex documentation structure, it can have its own `docs` directory under its root directory.
```

File: docs/documentation_guidelines.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+7. Each document file should have a prefix that reflects the purpose of the document, for example: `how-to_`, `proposal_`, `standard_`, `architecture_`. If a document covers multiple topics for example `how-to` and `architecture` the prefix is not necessary.
```
Comment (NHingerl) [inline docs/documentation_guidelines.md:9]: I'm on board with suffixes describing the purpose of the document.

However, I'm not sure how those "purposes" align with the [information types](https://kyma-project.io/community/guidelines/content-guidelines/01-content-strategy/#information-types) we have established for Kyma docs. We can certainly discuss whether the existing information types match modular Kyma (so far, discussions led to the conclusion that they do, perhaps with the addition of "architecture"). 

Also, mixing purposes in one document is not best practice in topic-based writing. 

In conclusion, this rule `7.` kind of competes with our content strategy; and while the strategy is absolutely open for discussion, we should indeed discuss any discrepancies first :) 

```suggestion
7. Each document file should have a prefix that reflects the purpose of the document, for example: `how-to_`, `proposal_`, `standard_`, `architecture_`. If a document covers multiple topics, such as `how-to` and `architecture`, the prefix is not necessary.
```

File: docs/documentation_guidelines.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+8. Main `docs` directory should contain structure to group `.md` files according to their purpose. For example there could be `proposal`, `how-to`, `architecture` directories and more.
```
Comment (NHingerl) [inline docs/documentation_guidelines.md:10]: This sentence might be good before ´4.´ :)

```suggestion
8. The main `docs` directory should contain a structure to group `.md` files according to their purpose. For example, there could be `proposal`, `how-to`, `architecture` directories and more.
```

File: docs/documentation_guidelines.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+9. An automated index page from all repositories should be linked to the [Kyma Organisation](https://github.tools.sap/kyma/documentation) central documentation.
```
Comment (NHingerl) [inline docs/documentation_guidelines.md:11]: Doesn't this kind of belong to ´1.`? Maybe move it higher up so that all information about the index page is consolidated.

**Review by NHingerl — COMMENTED** [review]
I reviewed this just to make sure we're within the 1 day timeline :)

I won't be working tomorrow but Iwona is there for any urgencies. I'm back on Monday.

File: docs/documentation_guidelines.md (hunk 6 above)
Comment (kasiakepka) [reply docs/documentation_guidelines.md]: Thanks Nina, it makes sense to have "one file per topic" so I decided to remove this part:

> If a document covers multiple topics for example `how-to` and `architecture` the prefix is not necessary.

Regarding prefixes, I am wondering if we have to strictly follow documentation guidelines for Kyma or new guidelines for modular Kyma since test-infra is not a part of a product.

Comment (kasiakepka) [reply docs/documentation_guidelines.md]: "Information types" listed [here](https://kyma-project.io/community/guidelines/content-guidelines/01-content-strategy#information-types) are good but we would be happy if we could  add more types like `architecture` or `standard`. What do you think, is that a crime? :)

Comment (kasiakepka) [reply docs/documentation_guidelines.md]: After discussion with Iwona I decided to get rid of document types  that are not mentioned in [information types](https://kyma-project.io/community/guidelines/content-guidelines/01-content-strategy/#information-types). I hope that we are now aligned 😄

File: docs/documentation_guidelines.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# Documentation guidelines
+
+1. Each repository should contain an automatically updated index page in `docs` directory.
+2. An automated index page from all repositories should be linked to the [Kyma Organisation](https://github.tools.sap/kyma/documentation) central documentation.
+3. The repository's main `README.md` file should contain a link to the repository's index page, where all documentation content is stored.
+4. The main `docs` directory should contain a structure to group `.md` files according to their purpose. For example, there could be `what-it`, `how-to`, `architecture` directories and more.
+5. Each `.md` file should begin with a title and in the paragraph below, a document description . These two elements of `.md` file will be used to create index page.
+6. Documentation for specific tools and libraries should be maintained in the same directory as the tool or library.
```
Comment (IwonaLanger) [inline docs/documentation_guidelines.md:8]:
```suggestion
6. Documentation for specific tools and libraries must be maintained in the same directory as the tool or library.
```

File: docs/documentation_guidelines.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+7. If a tool has a more complex documentation structure, it can have its own `docs` directory under its root directory.
+8. Each document file should have a prefix that reflects the purpose of the document, for example: `how-to_`, `what-is_`, `architecture_`.
```
Comment (IwonaLanger) [inline docs/documentation_guidelines.md:10]:
```suggestion
8. Each document file must have a prefix that reflects the purpose of the document, for example, `how-to_`, `what-is_`, or `architecture_`.
```

File: docs/documentation_guidelines.md [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+9. A `docs` directory under the repository root doesn't have to contain documentation related strictly to the one tool only. For example, we can have a documentation describing a system comprised of multiple tools which together deliver some service for users.
```
Comment (IwonaLanger) [inline docs/documentation_guidelines.md:11]:
```suggestion
9. A `docs` directory under a repository root doesn't have to contain documentation related strictly to one tool only. For example, there can be documentation describing a system comprised of multiple tools which together deliver one service for users.
```

**Review by IwonaLanger — APPROVED** [review]
